### ISSUE TITLE
Windows explorer frequently hangs

### ISSUE THREAD

**Issue by claell** [nextcloud/desktop#3451]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Expected behaviour
Nextcloud should seamlessly integrate

### Actual behaviour
I find the Windows Explorer hanging

### Steps to reproduce
1. Use Virtual Filesystem on Windows
2. Use a larger folder, versioned with git
3. Work inside that folder

### Client configuration
Client version: 3.2.2

Operating system: Windows 10

OS language: German

Qt version used by client package (Linux only, see also Settings dialog):

Client package (From Nextcloud or distro) (Linux only):

Installation path of client:

### Server configuration

Nextcloud version:

Storage backend (external storage):

### Logs

Please use Gist (https://gist.github.com/) or a similar code paster for longer
logs.

1. Client logfile: 
Since 3.1: Under the "General" settings, you can click on "Create Debug Archive ..." to pick the location of where the desktop client will export the logs and the database to a zip file.
On previous releases: Via the command line: `nextcloud --logdebug --logwindow` or `nextcloud --logdebug --logfile log.txt`
(See also https://docs.nextcloud.com/desktop/3.0/troubleshooting.html#log-files)

2. Web server error log:

3. Server logfile: nextcloud log (data/nextcloud.log):

**Comment by claell** [comment]
Currently, I am not sure what causes the problem. I assume that some of the points mentioned might be factors (like having a git repository synced and Virtual Filesystem enabled).

Possibly having the git repository open in VS Code causes problems where both VS Code and Nextcloud are checking files or something like that.

The explorer seems to try to refresh things, causing it to hang. This also applies to newly opened Windows of it.

**Comment by claell** [comment]
This has nothing to do with VS Code being open. I am currently experiencing the problem without that.

I have started the computer, used it for a while (did not access any files or opened a Windows explorer window). Now when I opened a Windows explorer window, it loads and loads (the content of the start display folder) and is not responsive.

A bit similar to the symptoms here (but worse in my experience): https://stackoverflow.com/questions/67824428/windows-explorer-has-problems-with-refreshing-folders-when-network-drive-is-offl.

**Comment by gaschka** [comment]
I don't use a virtual file system or git, but I'm dealing with large file sequences of PNG and EXR files.

**Comment by claell** [comment]
Might have been a duplicate of #3417 that got fixed with #3390. Not sure about that, though. Need to test first on the affected system.

**Comment by claell** [comment]
Push, currently we probably need feedback of the affected people whether the issue still occurs.

**Comment by gaschka** [comment]
> Push, currently we probably need feedback of the affected people whether the issue still occurs.

Hello, I still do have the impression that Nextcloud is causing lockups and slowdowns of the explorer, it's less worse.
I moved from using PNG to TIF sequences, perhaps this helped as well, as creating previews is much faster for Windows - so it's hard to say if it's the update.

**Comment by JensUweB** [comment]
I'm using nextcloud desktop version 3.4.0 (windows-10.0.19043).
I added a folder with approximately 12000 files (~512MiB in total) to the nextcloud folder to sync. The app just stopped working after a few seconds of trying to sync the files.
In the nextcloud app settings I had the download & upload bandwith limited to a fixed value. It seems, that this bandwith limitation causes the app to stop responding. After I disabled the bandwith limitation, the syncronisation of the 12000 files
worked well (but took some time). Virtual files are enabled.

Hope this helps someone.

**Comment by RedFox134** [comment]
> I'm using nextcloud desktop version 3.4.0 (windows-10.0.19043). I added a folder with approximately 12000 files (~512MiB in total) to the nextcloud folder to sync. The app just stopped working after a few seconds of trying to sync the files. In the nextcloud app settings I had the download & upload bandwith limited to a fixed value. It seems, that this bandwith limitation causes the app to stop responding. After I disabled the bandwith limitation, the syncronisation of the 12000 files worked well (but took some time). Virtual files are enabled.
> 
> Hope this helps someone.

This appears to have been the issue I was having as well.  The desktop app would freeze and become unresponsive within seconds of launching.  After disabling my upload limit the sync began and completed with no issues.  Glad it's working, but frustrating for those of us with monthly data caps.  I have another library I'd like to back up to NextCloud, but it's over 500GB of video files.  That'll take up a good chunk of my monthly bandwidth cap if I upload it all at once.  Limiting it was a way to spread out the data usage for me across a few months.  May have to see if it's an option to use a different connection while it uploads if I can't use the limit speed bandwidth options.

**Comment by godfuture** [comment]
> Not sure if Virtual Files is a new feature of Next Cloud, but it appears to be not ready for production.

Good point. In terms of quality, the Nextcloud client is one of the weak points. I have so many issues and ask myself, if others do not have the same issues. Being here in Github I guess I am not alone. Apparently the client is not ready for production in my opinion.

**Comment by Rello** [comment]
Hello,

this ticket is pretty old and the client had several fundamental redesigns.
If the issue still exists, please create a new ticket with current information.

Thank you
